src/core/namespace.c

```diff
@@ -189,6 +206,31 @@ static int append_protect_kernel_tunables(BindMount **p, const char *root_direct
                                     ELEMENTSOF(protect_kernel_tunables_table));
 }
 
+static int append_protect_home(BindMount **p, const char *root_directory, ProtectHome protect_home) {
+        int r = 0;
+
+        assert(p);
+
+        switch (protect_home) {
+        case PROTECT_HOME_READ_ONLY:
+                r = append_target_mounts(p, root_directory, protect_home_read_only_table,
+                                         ELEMENTSOF(protect_home_read_only_table));
+                break;
+        case PROTECT_HOME_YES:
+                r = append_target_mounts(p, root_directory, protect_home_yes_table,
+                                         ELEMENTSOF(protect_home_yes_table));
+                break;
+        case PROTECT_HOME_NO:
+                r = 0;
+                break;
+        default:
+                r = -EINVAL;
```

Comment:
IMO at this stage `assert_not_reached` is not good since we break hard, better use it when we load the properties... and here just fail with -EINVAL, but if you think that I should change it, I will add a separate patch on top.